scripts/encapp_tests.py

```diff
@@ -0,0 +1,390 @@
+#!/usr/local/bin/ python3
+import os
+import subprocess
+import json
+import sys
+import argparse
+import re
+from datetime import datetime
+from plot_rd import RDPlot
+
+KEY_NAME_DISCRIPTION="description"
+KEY_NAME_INPUT_FILES="input_files"
+KEY_NAME_CODECS="codecs"
+KEY_NAME_ENCODE_RESOLUTIONS="encode_resolutions"
+KEY_NAME_RC_MODES="rc_modes"
+KEY_NAME_BITRATES="bitrates"
+KEY_NAME_I_INTERVALS="i_intervals"
+KEY_NAME_DURATION="duration"
+KEY_NAME_USE_SURFACE_ENC="use_surface_enc"
+
+
+
+sample_config_json_data = [
+    {
+        KEY_NAME_DISCRIPTION: "sample",
+        KEY_NAME_INPUT_FILES: [""],
+        KEY_NAME_USE_SURFACE_ENC: 1,
+        KEY_NAME_CODECS: ["hevc"],
+        KEY_NAME_ENCODE_RESOLUTIONS: ["1280x720"],
+        KEY_NAME_RC_MODES: ["cbr"],
+        KEY_NAME_BITRATES: [500,1000,1500,2000,2500],
+        KEY_NAME_I_INTERVALS: [2,10,60],
+        KEY_NAME_DURATION: 10
+    }
+]
+TEST_CLASS_NAME="com.facebook.encapp.CodecValidationInstrumentedTest"
+JUNIT_RUNNER_NAME="com.facebook.encapp.test/android.support.test.runner.AndroidJUnitRunner"
+ENCAPP_OUTPUT_FILE_NAME_RE = '.*_[0-9]+fps_[0-9]+x[0-9]+_[0-9]+bps_iint[0-9]+_m[0-9]+(?:\.mp4$|\.webm)'
+RD_RESULT_FILE_NAME = "rd_results.json"
+
+def run_cmd(cmd):
+    try:
+        print(cmd, sep=' ')
+        process = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+        # print ('standard output: {}'.format(stdout.decode()))
+        # print ('standard error: {}'.format(stderr.decode()))
+    except:
+        raise Exception('Failed to run adb command')
+
+    return stdout.decode(), stderr.decode()
```

Comment:
return retcode too (so the caller can know whether anything failed)

Also, are you sure `process.communicate` wait3's ?

Reply:
Process.communicate() waits for the process to terminate